habitat_sim/sensors/noise_models/redwood_depth_noise_model.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import os.path as osp
+
+import attr
+import numba
+import numpy as np
+
+from habitat_sim.bindings import SensorType, cuda_enabled
+from habitat_sim.registry import registry
+from habitat_sim.sensors.noise_models.sensor_noise_model import SensorNoiseModel
+
+if cuda_enabled:
+    from habitat_sim._ext.habitat_sim_bindings import RedwoodNoiseModelGPUImpl
+    import torch
+
+
+@numba.jit(nopython=True)
+def _undistort(x, y, z, model):
+    i2 = int((z + 1) / 2)
+    i1 = int(i2 - 1)
+    a = (z - (i1 * 2.0 + 1.0)) / 2.0
+    x = x // 8
+    y = y // 6
+    f = (1 - a) * model[y, x, min(max(i1, 0), 4)] + a * model[y, x, min(i2, 4)]
+
+    if f < 1e-5:
+        return 0
+    else:
+        return z / f
+
+
+@numba.jit(nopython=True, parallel=True)
+def _simulate(gt_depth, model, noise_multiplier):
+    noisy_depth = np.empty_like(gt_depth)
+
+    H, W = gt_depth.shape
+    ymax, xmax = H - 1, W - 1
+
+    rand_nums = np.random.randn(H, W, 3).astype(np.float32)
+    for j in range(H):
+        for i in range(W):
+            y = int(
+                min(max(j + rand_nums[j, i, 0] * 0.25 * noise_multiplier, 0.0), ymax)
+                + 0.5
+            )
+            x = int(
+                min(max(i + rand_nums[j, i, 1] * 0.25 * noise_multiplier, 0.0), xmax)
+                + 0.5
+            )
+
+            # Downsample
+            d = gt_depth[y - y % 2, x - x % 2]
+            # If the depth is greater than 10, the sensor will just return 0
+            if d >= 10.0:
+                noisy_depth[j, i] = 0.0
+            else:
+                # Distort
+                undistorted_d = _undistort(
+                    int(x / xmax * 639.0 + 0.5), int(y / ymax * 479.0 + 0.5), d, model
```

Comment:
480x640 is the resolution of the sensor used to create the distortion model originally.  This logic just rescales any given observation to 480x640 so that it can query the distortion model.